lib/Conversion/TorchToTMTensor/TorchToTMTensor.cpp

```diff
@@ -0,0 +1,208 @@
+
+#include "torch-mlir/Conversion/TorchToTMTensor/TorchToTMTensor.h"
+
+#include "../PassDetail.h"
+#include "mlir/Dialect/Linalg/IR/Linalg.h"
+#include "mlir/Dialect/StandardOps/IR/Ops.h"
+#include "mlir/IR/MLIRContext.h"
+#include "torch-mlir-dialects/Dialect/TMTensor/IR/TMTensorDialect.h"
+#include "torch-mlir-dialects/Dialect/TMTensor/IR/TMTensorOps.h"
+#include "torch-mlir/Conversion/Utils/Utils.h"
+#include "torch-mlir/Dialect/Torch/IR/TorchDialect.h"
+#include "torch-mlir/Dialect/Torch/IR/TorchOps.h"
+#include "torch-mlir/Dialect/Torch/IR/TorchTypes.h"
+#include "torch-mlir/Dialect/Torch/Utils/TorchUpstream.h"
+#include "torch-mlir/Dialect/Torch/Utils/Utils.h"
+#include "torch-mlir/Dialect/TorchConversion/Transforms/BackendTypeConversion.h"
+
+using namespace mlir;
+using namespace mlir::torch;
+using namespace mlir::torch::Torch;
+using namespace mlir::torch::TorchConversion;
+using namespace mlir::torch::TMTensor;
+
+// -----------------------------------------------------------------------------
+// Patterns (as this grows, it should be organized into multiple files)
+// -----------------------------------------------------------------------------
+// This is going to eventually be O(#aten ops), which is in the 100s.
+//
+// Most of these patterns consist of:
+// 1. Checking that the operand/result types and other static properties are
+//    good-enough to create a valid linalg op (such as operands being of
+//    ranks/dtypes acceptable to the linalg op).
+// 2. Creating dynamic error guards, usually checking a predicate on the
+//    compatibility of operand shapes.
+// 3. Creating init tensors for the computation op. Usually this involves
+//    reifying IR for a shape transfer function based on the operand shapes.
+// 4. Creating a named linalg op to replace the original op.
+//
+// TODO: Use linalg OpDSL to autogenerate at least 1)/2)/3) such
+// that these patterns become mostly mechanical associations of
+// "aten.foo -> linalg.foo".
+
+namespace {
+// aten::bincount op counts the frequency of each value in a 1-d input tensor of
+// non-negative ints.
+class ConvertAtenBincountOp : public OpConversionPattern<AtenBincountOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenBincountOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+    Location loc = op.getLoc();
+    MLIRContext *context = op->getContext();
+    Value input = adaptor.self();
+    Value minlength = adaptor.minlength();
+    Value weights = adaptor.weights();
+
+    // Check whether the input is a 1-d tensor or not.
+    RankedTensorType inputType = input.getType().cast<RankedTensorType>();
+    if (inputType.getRank() != 1)
+      return rewriter.notifyMatchFailure(
+          op, "Input tensor has to be one dimensional.");
+
+    // TODO: Incorporate the weight argument.
```

Comment:
Can you add another `TODO` saying we also need to verify the input tensor elements are all positive.

Reply:
I don't think we need to add a guard for this. If we pass a tensor containing negative values as input, the python throws an error for this, saying:
```
RuntimeError: bincount only supports 1-d non-negative integral inputs.
```

Reply:
yea, the error you are seeing above is from the TorchScript (baseline) running with the invalid input. As a compiler, we need to guard the invalid code in our code base.